tests/ducktape/test_consumer.py

```diff
@@ -147,73 +185,245 @@ def test_basic_consume(self, consumer_type, batch_size):
+            topic_name = f"test-topic-changes-{uuid.uuid4()}"
+            group_id = f"topic-changes-group-{uuid.uuid4()}"  # Shared group ID
+
+            # Setup: Create topic with 2 partitions initially
+            self.kafka.create_topic(topic_name, partitions=2, replication_factor=1)
+            topic_ready = self.kafka.wait_for_topic(topic_name, max_wait_time=30)
+            assert topic_ready, f"Topic {topic_name} was not created"
+            self.produce_test_messages(topic_name, num_messages=10)
+
+            # Create consumers with shared group ID
+            consumer1 = self.create_consumer("async", group_id)
+            consumer2 = self.create_consumer("async", group_id)
+
+            # Track rebalance events
+            rebalance_events = []
+
+            async def track_rebalance(consumer, partitions):
+                rebalance_events.append(len(partitions))
+                await consumer.assign(partitions)
+
+            # Both consumers join - should get 1 partition each (2 total)
+            await consumer1.subscribe([topic_name], on_assign=track_rebalance)
+            await consumer2.subscribe([topic_name], on_assign=track_rebalance)
+
+            # Wait for initial rebalance
+            for attempt in range(10):
+                await consumer1.poll(timeout=1.0)
+                await consumer2.poll(timeout=1.0)
+
+                assignment1 = await consumer1.assignment()
+                assignment2 = await consumer2.assignment()
+
+                if len(assignment1) > 0 and len(assignment2) > 0:
+                    break
+                await asyncio.sleep(1.0)
+
+            # Verify initial state: 2 partitions total, 1 each
+            assignment1_initial = await consumer1.assignment()
+            assignment2_initial = await consumer2.assignment()
+            total_partitions_initial = len(assignment1_initial) + len(assignment2_initial)
+
+            assert total_partitions_initial == 2, \
+                f"Should have 2 total partitions initially, got {total_partitions_initial}"
+            assert len(rebalance_events) >= 2, \
+                f"Should have at least 2 rebalance events, got {len(rebalance_events)}"
+
+            # Add partitions to existing topic (2 -> 4 partitions)
+            self.kafka.add_partitions(topic_name, new_partition_count=4)
+
+            # Produce messages to new partitions to trigger metadata refresh
+            self.produce_test_messages(topic_name, num_messages=5)
+
+            # Force rebalance by creating a new consumer that joins the group
+            # This will trigger metadata refresh and rebalancing for all consumers
+            consumer3 = self.create_consumer("async", group_id)
+            await consumer3.subscribe([topic_name], on_assign=track_rebalance)
+
+            # Poll all consumers until they detect new partitions and rebalance
+            for _ in range(15):  # Max 15 seconds for partition discovery
```

Comment:
Might be longer given the consumers each poll as well for 1 second consecutively correct?